Foundation/NSURL.swift

```diff
@@ -858,9 +858,49 @@ public class NSURLComponents: NSObject, NSCopying {
     public override func copy() -> AnyObject {
         return copy(with: nil)
     }
-    
+
+    public override func isEqual(_ object: AnyObject?) -> Bool {
+        if let other = object as? NSURLComponents {
+            if scheme != other.scheme {
+                return false
+            }
+            if user != other.user {
+                return false
+            }
+            if password != other.password {
+                return false
+            }
+            if host != other.host {
+                return false
+            }
+            if port != other.port {
+                return false
+            }
+            if path != other.path {
+                return false
+            }
+            if query != other.query {
+                return false
+            }
+            if fragment != other.fragment {
+                return false
+            }
+            return true
+        }
+        return false
+    }
+
     public func copy(with zone: NSZone? = nil) -> AnyObject {
-        NSUnimplemented()
+        let copy = NSURLComponents()
+        copy.scheme = self.scheme
+        copy.user = self.user
+        copy.password = self.password
+        copy.host = self.host
+        copy.port = self.port
+        copy.path = self.path
+        copy.query = self.query
+        copy.fragment = self.fragment
+        return copy
```

Comment:
Again the same issue above applies to subclassers; @mbvreddy's approach is correct for the abstract implementation.